Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -12,19 +12,34 @@ public Pawn(Player player)
         public override IEnumerable<Square> GetAvailableMoves(Board board) {
             List<Square> available = new List<Square>();
             Square here = board.FindPiece(this);
-            if (Player == Player.White) {
-                if (here.Row != 0) {
-                    available.Add(new Square(here.Row - 1, here.Col ));
-                }
-                if (here.Row == 6) {
-                    available.Add(new Square(4, here.Col ));
+            if (Player == Player.White)
+            {
+                Square targetMove = new Square(here.Row - 1, here.Col);
+                if (targetMove.IsValid() && board.IsSquareEmpty(targetMove))
+                {
+                    Square targetMove2 = new Square(here.Row - 2, here.Col);
+                    if (here.Row != 0)
```

Comment:
Now that you are checking if target square is valid you do not need to check if new values is < 0